### ISSUE TITLE
Renders off screen if invoked far right or bottom

### ISSUE THREAD

**Issue by Risingson** [UnrealSecurity/context-js#3]
Thanks a lot for this library. A crucial thing missing is that it should check if it can render visibly on screen. If not the menu and its submenus should open to the left xor up.
![image](https://user-images.githubusercontent.com/5108701/106365773-0483d880-6338-11eb-9cf0-0060b2ecb922.png)

**Comment by heapoverride** [comment]
Agreed. I will try to work on this as soon as I can. Shouldn't take too long.

**Comment by kurtcodemander** [comment]
I added the following code at the bottom of method "show(x, y)" and it works fine for my use case. 

However, if there are several levels of submenus then they will open on top of each other (but can still be used). Ideally all submenus should probably open to the left of the main context menu when there is little room on the right...

```
let docWidth = document.documentElement.clientWidth, docHeight = document.documentElement.clientHeight; 

let dimensions = this.dom.getBoundingClientRect();
        
if (dimensions.left + dimensions.width > docWidth) {
    this.dom.style.left = docWidth - dimensions.width + 'px'; 
}

if (dimensions.top + dimensions.height > docHeight) {
    this.dom.style.top = docHeight - dimensions.height + 'px';
}  
```
